docker/Dockerfile

```diff
@@ -0,0 +1,137 @@
+# syntax=docker/dockerfile:1.6
+#
+# This Dockerfile is used to build the Llama Stack container image.
+# Example:
+# docker build \
+#   -f docker/Dockerfile \
+#   --build-arg DISTRO_NAME=starter \
+#   --tag llama-stack:starter .
+
+ARG BASE_IMAGE=python:3.12-slim
+FROM ${BASE_IMAGE}
+
+ARG UV_HTTP_TIMEOUT=500
+ENV UV_HTTP_TIMEOUT=${UV_HTTP_TIMEOUT}
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+WORKDIR /app
+
+RUN set -eux; \
```

Comment:
if we want to allow for overriding the base image, this is probably necessary in some form, but if we want to stick with 3.12 this can be as simple as: 

```Containerfile
  RUN apt-get update && \
      apt-get install -y --no-install-recommends \
          git gcc g++ && \
      rm -rf /var/lib/apt/lists/*

  RUN pip install --no-cache-dir uv
  ENV UV_SYSTEM_PYTHON=1
```

Either way this looks pretty good!